### PR TITLE
Ref/better plots

### PR DESCRIPTION
**Pull request by ozkanuysal**
**Here is example usage of new plots with parameters & thresholds.**

```
from gesund import validation

args = {
    'annotations_json_path': '/home/ozkan/gesund/test_data/object_detection/gesund_custom_format/gesund_custom_format_annotations_obj_det.json',
    'predictions': '/home/ozkan/gesund/test_data/object_detection/gesund_custom_format/gesund_custom_format_predictions_obj_det.json',
    'class_mappings': '/home/ozkan/gesund/test_data/object_detection/test_class_mappings.json',
    'problem_type': 'object_detection',
    'format': 'gesund_custom_format',
    'write_results_to_json': True,
    'plot_configs': {

        
        #--------------------------#classificaition--------------------------
        #  'class_distributions': {'metrics': ['normal','pneumonia'], 'threshold': 10}, 
        #  'blind_spot': {'class_type': ['Average','1','0']}, 
        #  'performance_by_threshold': {'graph_type': 'graph_1', 'metrics': ['F1', 'Sensitivity', 'Specificity', 'Precision','FPR', 'FNR'], 'threshold': 0.2}, 
        #  'roc': {'roc_class': ['normal', 'pneumonia']}, 
        #  'precision_recall': {'pr_class': ['normal','pneumonia']}, 
        #  'confidence_histogram': {'metrics': ['TP', 'FP'], 'threshold': 0.5}, 
        #  'overall_metrics': {'metrics': ['AUC','Precision'], 'threshold': 0.2} 

        #--------------------------OD--------------------------------------

        'mixed_plot': {'mixed_plot': ['map10', 'map50', 'map75'], 'threshold': 0.5}, 
        'top_misses': {'min_miou': 0.70, 'top_n': 10}, 
        'confidence_histogram': {'confidence_histogram_labels': ['TP', 'FP']}, 
        'classbased_table': {'classbased_table_metrics': ['precision', 'recall', 'f1'], 'threshold': 0.2}, 
        'overall_metrics': {'overall_metrics_metrics': ['map', 'mar'], 'threshold': 0.5}, 
        'blind_spot': {'blind_spot_Average': ['mAP@50', 'mAP@10','mAR@max=10','mAR@max=100'],'threshold': 0.5} 

        #-----------------------------SS-----------------------------------

        # 'violin_graph': {'metrics':['Acc', 'Spec','AUC'], 'threshold': 0.5}, 
        # 'plot_by_meta_data': {'meta_data_args': [ 'FalsePositive','Dice Score','mean Sensitivity','mean AUC','Precision','AverageHausdorffDistance','SimpleHausdorffDistance']}, 
        # 'overall_metrics': {'overall_args': ['mean AUC', 'fwIoU','mean Sensitivity']}, 
        # 'classbased_table': {'classbased_table_args': 0.5}, 
        # 'blind_spot': {'blind_spot_args': ['fwIoU','mean IoU','mean Sensitivity','mean Specificity','mean Kappa','mean AUC','']}, 
 
    }
}

result = validation.run_metrics(args)
plotting = validation.plotting_metrics(result, args)
```

### CODE REVIEW

**Review by HammadK44 — APPROVED** [review]

File: gesund/metrics/classification/create_validation.py [hunk 1]
```diff
@@ -127,45 +127,63 @@ def load(self, validation_collection_data, class_mappings, filtering_meta=None):
         
         return overall_metrics
     
-    def plot_metrics(self, metrics, jsons_dir, plot_dir):
+    def plot_metrics(self, metrics, json_outputs_dir, plot_outputs_dir, plot_configs):
```
Comment (Xnsam) [inline gesund/metrics/classification/create_validation.py:130]: please lets add variable typing as well for better code clarity 
https://realpython.com/python-type-checking/

File: gesund/validation.py [hunk 2]
```diff
@@ -91,7 +91,7 @@ def run_metrics(args):
 
     return metrics
 
-def plotting_metrics(metrics, filtering_meta=None):
+def plotting_metrics(metrics, args, filtering_meta=None):
```
Comment (Xnsam) [inline gesund/validation.py:94]: i suggest, we could pass explicitly `args["plot_configs"]`, so that there is no need for `args.get`

**Review by Xnsam — CHANGES_REQUESTED** [review]

File: gesund/metrics/classification/create_validation.py (hunk 1 above)
Comment (ozkanuysal) [reply gesund/metrics/classification/create_validation.py]: thanks updated.

File: gesund/validation.py (hunk 2 above)
Comment (ozkanuysal) [reply gesund/validation.py:94]: I tested like replaced args.get('plot_configs', {}) with args['plot_configs'] in the function, it gives the same results.

**Pull request merged**